cmd/compute-domain-kubelet-plugin/driver.go

```diff
@@ -38,63 +40,64 @@ const ErrorRetryMaxTimeout = 45 * time.Second
 
 // permanentError defines an error indicating that it is permanent.
 // By default, every error will be retried up to ErrorRetryMaxTimeout.
-// Errors marked as permament will not be retried.
+// Errors marked as permanent will not be retried.
 type permanentError struct{ error }
 
 func isPermanentError(err error) bool {
 	return errors.As(err, &permanentError{})
 }
 
-var _ drapbv1.DRAPluginServer = &driver{}
-
 type driver struct {
 	sync.Mutex
 	client coreclientset.Interface
-	plugin kubeletplugin.DRAPlugin
+	helper *kubeletplugin.Helper
 	state  *DeviceState
 }
 
 func NewDriver(ctx context.Context, config *Config) (*driver, error) {
-	driver := &driver{
-		client: config.clientsets.Core,
-	}
-
 	state, err := NewDeviceState(ctx, config)
 	if err != nil {
 		return nil, err
 	}
-	driver.state = state
 
-	plugin, err := kubeletplugin.Start(
+	driver := &driver{
+		client: config.clientsets.Core,
+		state:  state,
+	}
+
+	driver.helper, err = kubeletplugin.Start(
```

Comment:
nit: could we write use:
```suggestion
	helper, err = kubeletplugin.Start(
```
and asign `driver.helper` after we have checked error. We could even move the instaniation of `driver` to just before we return -- although this is not a hard requirement from my end.

Reply:
doing that